draft-ietf-quic-transport.md

```diff
@@ -2606,6 +2606,16 @@ An endpoint sends a closing frame (CONNECTION_CLOSE or APPLICATION_CLOSE) to
 terminate the connection immediately.  Any closing frame causes all streams to
 immediately become closed; open streams can be assumed to be implicitly reset.
 
+If the endpoint has successfully decrypted a 1-RTT packet from its peer, it
+SHOULD send CONNECTION_CLOSE in a 1-RTT packet. If not, and it has received a
```

Comment:
This does not work if the client decides to abort the handshake in response to the server's Handshake messages, because the server's messages typically end with NewSessionTicket carried by a 1-RTT packet but the server cannot decrypt a 1-RTT packet from client until it receives ClientFinished.

IMO, we should state that an endpoint SHOULD send CONNECTION_CLOSE using a Handshake packet until either of the following to condition is met: a) it knows that the peer can decrypt a 1-RTT packet, b) it drops the Handshake keys based on the timer.

Other than that, the PR looks good to me.

Reply:
How about "the handshake is complete and it knows the peer can decrypt a 1-RTT packet."?

Reply:
That works for me, though I would prefer dropping the former (i.e. "handshake is complete") because it's confusing; IIUC we do not have a definition of when the handshake becomes complete, and we could argue in various ways on when it becomes complete. Actually, the latter condition (i.e. "knows the peer can decrypt a 1-RTT packet") is when the handshake is _known_ to have been complete.

Reply:
FWIW, a client "knows that the peer can decrypt a 1-RTT packet" when it receives the first ACK for a 1-RTT packet it has sent. A server knows that when it obtains the 1-RTT read key from the TLS stack (i.e. when a ClientFinished received from the client is processed without an error).

We might want to clarify that.